src/Debug/skeletonViewer.ts

```diff
@@ -387,16 +387,25 @@ export class SkeletonViewer {
         options.displayOptions.sphereScaleUnit = options.displayOptions.sphereScaleUnit ?? 2;
         options.displayOptions.sphereFactor = options.displayOptions.sphereFactor ?? 0.865;
         options.computeBonesUsingShaders = options.computeBonesUsingShaders ?? true;
+        options.useAllBones = options.useAllBones ?? true;
 
         const boneIndices = mesh.getVerticesData(VertexBuffer.MatricesIndicesKind);
         const boneWeights = mesh.getVerticesData(VertexBuffer.MatricesWeightsKind);
         this._boneIndices = new Set();
 
-        if (boneIndices && boneWeights) {
-            for (let i = 0; i < boneIndices.length; ++i) {
-                const index = boneIndices[i], weight = boneWeights[i];
-                if (weight !== 0) {
-                    this._boneIndices.add(index);
+        if (options.useAllBones) {
+            if (boneIndices) {
+                for (let i = 0; i < boneIndices.length; ++i) {
+                    this._boneIndices.add(boneIndices[i]);
+                }
+            }
+        }else {

```

Comment:
No need to fill the `this._boneIndices` set if `options.useAllBones` as it won't be used anyway.

Reply:
The bones were not appearing if not filled

Reply:
Simply do:
```javascript
if (!options.useAllBones) {
    if (initialMeshBoneIndices && initialMeshBoneWeights) {
        for (let i = 0; i < initialMeshBoneIndices.length; ++i) {
            const index = initialMeshBoneIndices[i], weight = initialMeshBoneWeights[i];
            if (weight !== 0) {
                this._boneIndices.add(index);
            }
        }
    }
}
```
and see my other comments below.